accelerator/proxy_apis.c

```diff
@@ -356,3 +365,36 @@ status_t api_remove_neighbors(const iota_client_service_t* const service, const
   char_buffer_free(res_buff);
   return ret;
 }
+
+status_t api_proxy_apis(const iota_client_service_t* const service, const char* const obj, char** json_result) {
+  status_t ret = SC_OK;
+  const uint8_t max_cmd_len = 30;
+  char command[max_cmd_len];
+
+  proxy_apis_command_req_deserialize(obj, command);
+  // TODO more proxy APIs should be implemented
+  if (strncmp(command, "checkConsistency", strlen("checkConsistency")) == 0) {
```

Comment:
Can you avoid repeated if-else statements? Use the techniques to switch with computed hash: https://stackoverflow.com/questions/4014827/best-way-to-switch-on-a-string-in-c/4014900

Reply:
The previous relating discussion in this PR. https://github.com/DLTcollab/tangle-accelerator/pull/335#discussion_r338540072
Be careful of collisions when using hash.

Reply:
Any further progress about string-based dispatching?

Reply:
I am using djb2 as hash function here.
I will push the new commit this afternoon

Reply:
You can add hash and related string-based dispatching routines into directory `utils/`.

Reply:
OK